packages/core/src/indexing/local-index-api.ts

```diff
@@ -186,7 +209,13 @@ export class LocalIndexApi implements IndexApi {
    */
   async populateDatabaseApiInternalState(): Promise<void> {
     const modelsToIndex = this.databaseIndexApi?.getActiveModelsToIndex()
```

Comment:
can we just update `getActiveModelsToIndex()` to give us all the information we need (or add a new function that gives us the info we need in a single call if `getActiveModelsToIndex` is used in other places that don't need the extra info).  That way we don't have to call into the databaseIndexApi multiple times back to back?